src/client/metadataApiDeploy.ts

```diff
@@ -374,8 +374,8 @@ export class MetadataApiDeploy extends MetadataTransfer<
   // eslint-disable-next-line @typescript-eslint/require-await
   protected async post(result: MetadataApiDeployStatus): Promise<DeployResult> {
     const lifecycle = Lifecycle.getInstance();
+    const connection = await this.getConnection();
```

Comment:
Why'd you move this outside the try?

Reply:
it's used below, outside the try

Reply:
Is that the right thing to do.  getting a connection seems like a path where an error could be thrown. Was wondering if we shouldn't just get a new connection if we need it below. 
...
but then I took a look at the method and saw we're just eating the exception anyway. :sadness: carry on

Reply:
logic: if you got to the post stage you definitely had a valid connection (otherwise, how'd you deploy and poll for completion?). Maybe something weird happens between, but probably not.

error swallowing: we don't want to throw on telemetry reporting.

Reply:
If the try/catch is only for telemetry should we put the try on line 392?